Windows agent fails to compile with MinGW version 10

|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.0 | Wazuh FIM | Agent | Sources | Windows |

While attempting to compile the Windows agent under a Debian distro that uses MinGW version 10, it fails with the following error messages:
```
/usr/bin/i686-w64-mingw32-ld: win32/win_service.o:win_service.c:(.bss+0x0): multiple definition of `isVista'; win32/win_agent.o:win_agent.c:(.bss+0x4): first defined here
/usr/bin/i686-w64-mingw32-ld: win32/win_utils.o:win_utils.c:(.bss+0x8): multiple definition of `wm_sca_sub_tree'; win32/win_agent.o:win_agent.c:(.bss+0x0): first defined here
/usr/bin/i686-w64-mingw32-ld: win32/win_utils.o:win_utils.c:(.bss+0xc): multiple definition of `isVista'; win32/win_agent.o:win_agent.c:(.bss+0x4): first defined here
```

Upon further inspection, we see that the `isVista` variable is defined inside `file_op.h` which later gets included into several files, causing the variable to show up as defined multiple times.

https://github.com/wazuh/wazuh/blob/89aa5cbe466b74f16fc195402dea9b36742103a6/src/headers/file_op.h#L33

We should move this variable definition into `file_op.c` and add the `extern` qualifier in the header file.

--- a/src/Makefile
+++ b/src/Makefile
@@ -1178,7 +1178,7 @@ wazuh_modules/%.o: wazuh_modules/%.c
 
 ifeq (${TARGET},winagent)
 wazuh_modules/syscollector/syscollector_win_ext.dll: wazuh_modules/syscollector/syscollector_win_ext.o
-	${OSSEC_SHARED} ${OSSEC_LDFLAGS} $^ -o $@ $(OSSEC_LIBS) $(JSON_LIB) -liphlpapi -lws2_32
+	${OSSEC_SHARED} ${OSSEC_LDFLAGS} $^ -o $@ $(OSSEC_LIBS) $(JSON_LIB) -liphlpapi -lws2_32 -static-libgcc
 endif # DLL for Windows
 
 ifeq (,$(filter ${DISABLE_SYSC},YES yes y Y 1))

--- a/src/headers/file_op.h
+++ b/src/headers/file_op.h
@@ -30,7 +30,7 @@
 #ifdef WIN32
 #define PATH_SEP '\\'
 typedef uint64_t wino_t;
-int isVista;
+extern int isVista;
 #else
 #define PATH_SEP '/'
 typedef ino_t wino_t;

--- a/src/shared/file_op.c
+++ b/src/shared/file_op.c
@@ -387,6 +387,10 @@
 #define mkdir(x, y) mkdir(x)
 #endif /* WIN32 */
 
+#ifdef WIN32
+int isVista;
+#endif
+
 const char *__local_name = "unset";
 
 /* Set the name of the starting program */

--- a/src/wazuh_modules/wm_sca.c
+++ b/src/wazuh_modules/wm_sca.c
@@ -37,6 +37,10 @@ typedef struct request_dump_t {
     int first_scan;
 } request_dump_t;
 
+#ifdef WIN32
+static HKEY wm_sca_sub_tree;
+#endif
+
 static const int RETURN_NOT_FOUND = 0;
 static const int RETURN_FOUND = 1;
 static const int RETURN_INVALID = 2;

--- a/src/wazuh_modules/wm_sca.h
+++ b/src/wazuh_modules/wm_sca.h
@@ -29,10 +29,6 @@
 #define WM_SCA_STAMP          "sca"
 #define WM_CONFIGURATION_ASSESSMENT_DB_DUMP                   "sca-dump"
 
-#ifdef WIN32
-HKEY wm_sca_sub_tree;
-#endif
-
 typedef struct wm_sca_policy_t {
     unsigned int enabled:1;
     unsigned int remote:1;

--- a/src/win32/ui/common.c
+++ b/src/win32/ui/common.c
@@ -15,6 +15,8 @@
 #include "os_net/os_net.h"
 #include "validate_op.h"
 
+/* Agent status */
+char ui_server_info[2048 + 1];
 
 /* Generate server info (for the main status) */
 int gen_server_info(HWND hwnd)

--- a/src/win32/ui/os_win32ui.c
+++ b/src/win32/ui/os_win32ui.c
@@ -13,6 +13,8 @@
 #include "os_win32ui.h"
 #include "../os_win.h"
 
+ossec_config config_inst;
+HWND hStatus;
 
 /* Dialog -- About OSSEC */
 BOOL CALLBACK AboutDlgProc(HWND hwnd, UINT Message,

--- a/src/win32/ui/os_win32ui.h
+++ b/src/win32/ui/os_win32ui.h
@@ -69,14 +69,11 @@ typedef struct _ossec_config {
 
 /** Global variables **/
 
-/* Agent status */
-char ui_server_info[2048 + 1];
-
 /* Configuration */
-ossec_config config_inst;
+extern ossec_config config_inst;
 
 /* Status bar */
-HWND hStatus;
+extern HWND hStatus;
 
 /* Ossec icon */
 #define IDI_OSSECICON  201

